test/features/thread/domain/usecases/search_email_interactor_test.dart

```diff
@@ -0,0 +1,98 @@
+import 'package:dartz/dartz.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_filter_condition.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:model/model.dart';
+import 'package:tmail_ui_user/features/thread/domain/model/search_email.dart';
+import 'package:tmail_ui_user/features/thread/domain/repository/thread_repository.dart';
+import 'package:tmail_ui_user/features/thread/domain/state/search_email_state.dart';
+import 'package:tmail_ui_user/features/thread/domain/usecases/search_email_interactor.dart';
+
+import '../../../../fixtures/account_fixtures.dart';
+import '../../../../fixtures/session_fixtures.dart';
+import 'search_email_interactor_test.mocks.dart';
+
+@GenerateNiceMocks([MockSpec<ThreadRepository>()])
+void main() {
+  final threadRepository = MockThreadRepository();
+  final searchEmailInteractor = SearchEmailInteractor(threadRepository);
+
+  group('search email interactor test:', () {
```

Comment:
also write unit test for case: SearchSnippet failed , list email search are list email of `Email/get`

Reply:
Done